server/src-lib/Hasura/RQL/DDL/RemoteRelationship/Validate.hs

```diff
@@ -405,19 +405,25 @@ validateType permittedVariables value expectedGType types =
                         (G.toGT $ G.NamedType name)
                         "not an input object type"))
 
-assertType :: G.GType -> G.GType -> Validation (NonEmpty ValidationError) ()
-assertType actualType expectedType = do
-  -- check if both are list types or both are named types
-  (when
-     (isListType' actualType /= isListType' expectedType)
-     (Failure (pure $ ExpectedTypeButGot expectedType actualType)))
-  -- if list type then check over unwrapped type, else check base types
-  if isListType' actualType
-    then assertType (unwrapTy actualType) (unwrapTy expectedType)
-    else (when
-            (getBaseTy actualType /= getBaseTy expectedType)
-            (Failure (pure $ ExpectedTypeButGot expectedType actualType)))
-  pure ()
+isTypeCoercible :: G.GType -> G.GType -> Validation (NonEmpty ValidationError) ()
+isTypeCoercible actualType expectedType =
+  -- The GraphQL spec says that, a singleton type can be coerced into  an array
```

Comment:
I think technically "singleton" type is called `named` type but we can fix this comment later.